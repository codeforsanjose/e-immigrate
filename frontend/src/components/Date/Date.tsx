import React from 'react';
import './Date.css';
import { ErrorMessageOnlyContentText } from '../../types/ContentText';
import { CommonComponentProps } from '../../types/CommonComponentProps';

type DateProps = CommonComponentProps<ErrorMessageOnlyContentText>;
export function Date(props: DateProps) {
    const {
        q, bindField, content,
    } = props;
    return (
        <>
            <input
                type="date"
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)} />
            <div className="RequiredError">*{content.errorMessage}</div>
        </>
    );
}
