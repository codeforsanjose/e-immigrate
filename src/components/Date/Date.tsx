import React from 'react';
import './Date.css';
import { BindFieldFunction, ContentData, QData } from '../../types/common';

type DateProps = {
    attributes: {

        q: QData;
        bindField: BindFieldFunction;
        content: ContentData;
    };
}
export function Date(props: DateProps) {
    const {
        attributes: { q, bindField, content },
    } = props;
    return (
        <>
            <input
                type="date"
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)} />
            <div className="RequiredError">*{content.errorMessage}</div>
        </>
    );
}
