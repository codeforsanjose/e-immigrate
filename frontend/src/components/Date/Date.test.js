import React from 'react';
import Date from './Date';

import renderer from 'react-test-renderer';

it('Date renders correctly', () => {
    const tree = renderer
        .create(
            <Date
                q={{}}
                answers={[]}
                values={[]}
                bindField={() => {}}
                showFollowUp={{}}
                setShowFollowUp={{}}
                collectAnswer={{}}
                content={{}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
