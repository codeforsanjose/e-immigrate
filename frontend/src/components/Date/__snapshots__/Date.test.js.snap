// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date renders correctly 1`] = `
Array [
  <input
    className="TextInput"
    type="date"
  />,
  <div
    className="RequiredError"
  >
    *
  </div>,
]
`;
